ext/mbstring/mbstring.c

```diff
@@ -538,6 +538,7 @@ static void _php_mb_free_regex(void *opaque)
 }
 /* }}} */
 
+// TODO make void as return value is ignored?
 /* {{{ php_mb_nls_get_default_detect_order_list */
 static int php_mb_nls_get_default_detect_order_list(enum mbfl_no_language lang, enum mbfl_no_encoding **plist, size_t *plist_size)
 {
```

Comment:
@alexdowad what is your opinion on this?

Reply:
The boolean (`int` is wrong) return value indicates whether the needed data was found for language `lang` or not. The only caller ignores that status return and assumes that it always succeeds.

And indeed, it seems to me that there is no case when the call will not succeed. The caller ensures that it is never called with an invalid language, and for every valid language, I believe the "identify list" data is present. So the call will never fail.

Since that is so, I agree, you could kindly change the return type to `void`, then take the current `return 1` and change it to just `return`... but *also*, take the current `return 0` and change that to `ZEND_ASSERT(false)`.

Reply:
So it seems this can actually fail, I need to go for tonight but ``ext/mbstring/tests/mb_language.phpt`` is one of those tests.

Reply:
@Girgias I agree that changing `mbstring.language` can fail, but if so, `php_mb_nls_get_default_detect_order_list` should never be called. The code which catches an invalid value for `mbstring.language` is above the call, and the call should never be reached in that case.